src/fights.ts

```diff
@@ -351,6 +358,26 @@ function pygmyOptions(forceEquip: Item[] = []) {
   };
 }
 
+let _bestShadowRift: Location | null = null;
+export function bestShadowRift(): Location {
+  if (!_bestShadowRift) {
+    _bestShadowRift =
+      chooseRift({
+        canAdventure: true,
+        sortBy: (l: Location) => {
+          const drops = getMonsters(l)
+            .map((m) => Object.keys(itemDrops(m)).map((s) => toItem(s)))
+            .flat();
+          return sum(drops, garboValue);
+        },
+      }) ?? $location.none;
```

Comment:
```suggestion
      });
```
I think you just missed this part